src/Symfony/Component/Mailer/Bridge/Mailgun/Http/Api/MailgunTransport.php

```diff
@@ -27,15 +28,13 @@
  */
 class MailgunTransport extends AbstractApiTransport
 {
-    private const ENDPOINT = 'https://api.mailgun.net/v3/%domain%/messages';
-
     private $key;
-    private $domain;
+    private $endpoint;
 
-    public function __construct(string $key, string $domain, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
+    public function __construct(string $key, string $domain, string $region = MailgunRegionConfiguration::REGION_DEFAULT, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
     {
         $this->key = $key;
-        $this->domain = $domain;
+        $this->endpoint = MailgunRegionConfiguration::resolveApiEndpoint($domain, $region);
```

Comment:
I don't think the `MailgunRegionConfiguration` class is really needed. 

You can, just change the `ENDPOINT` constant in each Mailgun Transports (or add it for the smtp transport) to add the region endpoint, something like : `private const ENDPOINT = 'https://%region_endpoint%.mailgun.net/v3/%domain%/messages'`
And store the region the same way as the domain, `$this->regionEndpoint = 'us' === $region ? 'api' : 'api.eu';`

Then (at line 51) replace the domain and the region endpoint at the same time
`$endpoint = str_replace(['%domain%', '%region_endpoint%'], [urlencode($this->domain), $this->regionEndpoint], self::ENDPOINT);`

What do you think ?

Reply:
that might be become tricky as soon as new region for example Asia might have host name https://mailgun.asia/

Me too don't like that helper class but we have to resolve endpoint in 3 different transports so sooner or later we have met case at which that will require to be extracted

Reply:
I think they would use the same pattern instead of using something different. But anyway, that's speculation at this point. I prefer #31998, which implements it the same as what we have in `SesTransport`, so it's more consistent across the board.